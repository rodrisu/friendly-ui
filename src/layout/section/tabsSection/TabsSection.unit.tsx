import React from 'react'
import renderer from 'react-test-renderer'

import { TabsSection } from './index'

const tabs = {
  activeTabId: 'tab1',
  tabs: [
    {
      id: 'tab1',
      label: 'Tab label 1',
      panelContent: <div>panel 1</div>,
      badgeContent: 'Badge content 1',
    },
    {
      id: 'tab2',
      label: 'Tab label 2',
      panelContent: <div>panel 2</div>,
      badgeContent: '2',
      badgeAriaLabel: 'Unread Message',
    },
  ],
}

describe('TabsSection', () => {
  it('should render default TabsSection', () => {
    const section = <TabsSection tabsProps={tabs} />
    const renderedSection = renderer.create(section).toJSON()
    expect(renderedSection).toMatchSnapshot()
  })
})
