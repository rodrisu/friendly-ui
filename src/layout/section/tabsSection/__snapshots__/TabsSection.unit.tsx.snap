// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsSection should render default TabsSection 1`] = `
.c2 {
  box-sizing: border-box;
  display: inline-block;
  min-width: 18px;
  height: 18px;
  padding: 0 4px;
  font-size: 12px;
  line-height: 16px;
  font-weight: 400;
  text-align: center;
  border-radius: 18px;
  border: 1px solid #FFF;
  background-color: #F53F5B;
  color: #FFF;
}

.c1 {
  position: relative;
}

.c1 .kirk-tabs {
  border-bottom: 1px solid #DDD;
}

.c1 .kirk-tablist {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  overflow: auto;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c1 .kirk-tablist::-webkit-scrollbar {
  display: none;
}

.c1 .kirk-tab {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  outline: none;
  padding: 16px;
  background: none;
  border: none;
  background-color: #FFF;
  color: #00AFF5;
  line-height: normal;
  font-size: 18px;
}

.c1 .kirk-tab-content {
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  white-space: nowrap;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 .kirk-tab-content > .kirk-icon {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1 .kirk-tab-text {
  text-overflow: ellipsis;
  overflow: hidden;
}

.c1 .kirk-tab-text--with-icon {
  margin-left: 16px;
  text-align: left;
}

.c1 .kirk-tab-second-line {
  margin-top: 8px;
  max-width: 100%;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c1 .kirk-tab-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: last baseline;
  -webkit-box-align: last baseline;
  -ms-flex-align: last baseline;
  align-items: last baseline;
}

.c1 .kirk-tabs-fixed .kirk-tab-content {
  white-space: normal;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 .kirk-tab-selected .kirk-tab {
  color: #054752;
}

.c1 .kirk-tab:not(.kirk-tab-selected):hover {
  cursor: pointer;
}

.c1 .kirk-tab-container .kirk-badge {
  position: relative;
  left: 0;
  top: calc(-1 * 4px);
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c1 .kirk-tab-highlight {
  position: absolute;
  z-index: 2;
  left: 0;
  bottom: 0;
  width: 0;
  height: 2px;
  background-color: #054752;
  -webkit-transition-property: width,left;
  transition-property: width,left;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

@media (min-width:800px) {
  .c0 .kirk-tablist-wrapper {
    width: calc(662px - 2 * 24px);
    margin: 0 auto;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    role="presentation"
  >
    <div
      className="kirk-tabs"
    >
      <div
        className="kirk-tablist-wrapper"
      >
        <div
          className="kirk-tab-highlight"
        />
        <div
          aria-multiselectable="false"
          aria-orientation="horizontal"
          className="kirk-tablist"
          role="tablist"
        >
          <div
            className="kirk-tab-container kirk-tab-selected"
            style={null}
          >
            <button
              aria-controls="tab1_panel"
              aria-selected="true"
              className="kirk-tab"
              id="tab1"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="tab"
              tabIndex={0}
              title="Tab label 1"
            >
              <span
                className="kirk-tab-content"
              >
                <span
                  className="kirk-tab-text"
                >
                  Tab label 1
                </span>
                <span
                  className="c2 kirk-badge kirk-tab-badge"
                >
                  <span
                    aria-hidden={false}
                  >
                    Badge content 1
                  </span>
                </span>
              </span>
            </button>
          </div>
          <div
            className="kirk-tab-container"
            style={null}
          >
            <button
              aria-controls="tab2_panel"
              aria-selected="false"
              className="kirk-tab"
              id="tab2"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="tab"
              tabIndex={-1}
              title="Tab label 2 Unread Message"
            >
              <span
                className="kirk-tab-content"
              >
                <span
                  className="kirk-tab-text"
                >
                  Tab label 2
                </span>
                <span
                  aria-label="Unread Message"
                  className="c2 kirk-badge kirk-tab-badge"
                >
                  <span
                    aria-hidden={true}
                  >
                    2
                  </span>
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="kirk-panels-section"
      role="presentation"
    >
      <div
        aria-labelledby="tab1"
        className="kirk-tab-panel"
        hidden={false}
        id="tab1_panel"
        role="tabpanel"
      >
        <div>
          panel 1
        </div>
      </div>
      <div
        aria-labelledby="tab2"
        className="kirk-tab-panel"
        hidden={true}
        id="tab2_panel"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;
